aibolit/patterns/method_chaining/method_chaining.py

```diff
@@ -1,68 +1,47 @@
-                            new_uuid = uuid.uuid1()
-                            dict_with_chains[new_uuid].append([j.position.line, j.member])
-                            self.traverse_node(j, dict_with_chains, new_uuid)
-
-                    elif isinstance(j, javalang.tree.MethodDeclaration):
-                        self.traverse_node(j, dict_with_chains, str(uuid.uuid1()))
-
-                    elif isinstance(j, javalang.tree.StatementExpression):
-                        self.traverse_node(j, dict_with_chains, uuid_method)
-
-                    elif isinstance(j, javalang.tree.This) or isinstance(j, javalang.tree.ClassCreator):
-                        self.traverse_node(j, dict_with_chains, str(uuid.uuid1()))
-            elif isinstance(item, javalang.tree.ClassCreator):
-                self.traverse_node(item, dict_with_chains, uuid_method)
-
-        return dict_with_chains
-
-
-    # flake8: noqa: C901
-    def value(self, filename: str):
-        """
-        Travers over AST tree finds method chaining. It is searched in a statement
-        :param filename:
-        :return:
-        List of tuples with LineNumber and List of methods names, e.g.
-        [[10, 'func1'], [10, 'fun2']], [[23, 'run'], [23, 'start']]]
-        """
-        tree = build_ast(filename)
-        chain_lst = defaultdict(list)
-        for _, node in tree.filter(javalang.tree.StatementExpression):
-            if isinstance(node.expression, javalang.tree.MethodInvocation):
-                children = node.children
-                if isinstance(children[1], javalang.tree.MethodInvocation):
-                    uuid_first_method = str(uuid.uuid1())
-                    chain_lst[uuid_first_method].append([
-                        children[1].position.line, children[1].member])
-                    self.traverse_node(children[1], chain_lst, uuid_first_method)
-
-        filtered_dict = list(filter(lambda elem: len(elem) > 1, chain_lst.values()))
-        return [item[0][0] for item in filtered_dict]
+    '''
+    In this pattern we check whether
+    more than one method chaining
+    invocation is used or not.
+    '''
+    def _is_invocation_inside(self, node: ASTNode) -> bool:
+        # to process structures such as:
+        # obj.methodInv1(list).methodInv2("blabla")
+        for i in node.children:
+            if i.node_type == ASTNodeType.METHOD_INVOCATION:
+                return True
+
+        return False
+
+    def _check_chained_method(self, ast: AST, method_invocation: ASTNode) -> bool:
+        childs = 0
+        for node in method_invocation.children:
+            if node.node_type == ASTNodeType.METHOD_INVOCATION:
+                # we check structures such as:
+                # new MyObject().Start()
+                #   .SpecifySomeParameter(list)
+                #   .SpecifySomeOtherParameter("list")
+                if self._is_invocation_inside(node) or node.qualifier is None:
```

Comment:
This two `if` could be one.

Reply:
Also leave a comment about `node.qualifier is None`. Why do we do this check?

Reply:
because this way we find method chaining

a().b() ->b has  node.qualifier None and we have invocation here